includes/ReverseWithdrawal/Manager.php

```diff
@@ -0,0 +1,782 @@
+        if ( ! empty( $args['trn_date']['from'] ) && ! empty( $args['trn_date']['to'] ) ) {
+            // fix min max value
+            if ( $args['trn_date']['from'] > $args['trn_date']['to'] ) {
+                $temp = $args['trn_date']['from'];
+                $args['trn_date']['from'] = $args['trn_date']['to'];
+                $args['trn_date']['to'] = $temp;
+            }
+
+            $where        .= ' AND trn.trn_date BETWEEN %d AND %d';
+            $query_args[] = $args['trn_date']['from'];
+            $query_args[] = $args['trn_date']['to'];
+        } elseif ( ! empty( $args['trn_date']['to'] ) ) {
+            $where        .= ' AND trn.trn_date <= %d';
+            $query_args[] = $args['trn_date']['to'];
+        } elseif ( ! empty( $args['trn_date']['from'] ) ) {
+            $where        .= ' AND trn.trn_date >= %d';
+            $query_args[] = $args['trn_date']['from'];
+        }
+
+        // order and order by param
+        // supported order by param
+        $supported_order_by = [
+            'id'        => 'trn.id',
+            'trn_date'  => 'trn.trn_date',
+            'debit'     => 'trn.debit',
+            'credit'    => 'trn.credit',
+        ];
+
+        if ( ! empty( $args['orderby'] ) && array_key_exists( $args['orderby'], $supported_order_by ) ) {
+            $order   = in_array( strtolower( $args['order'] ), [ 'asc', 'desc' ], true ) ? strtoupper( $args['order'] ) : 'ASC';
+            $orderby = "ORDER BY {$supported_order_by[ $args['orderby'] ]} {$order}"; //no need for prepare, we've already whitelisted the parameters
+
+            //second order by in case of similar value on first order by field
+            if ( 'id' !== $args['orderby'] ) {
+                $orderby .= ", trn.id {$order}";
+            }
+        }
+
+        // pagination param
+        if ( ! empty( $args['per_page'] ) && -1 !== intval( $args['per_page'] ) && ! in_array( $args['return'], [ 'count', 'balance_count', 'vendor_transaction_count' ], true ) ) {
+            $limit  = absint( $args['per_page'] );
+            $page   = absint( $args['page'] );
+            $page   = $page > 0 ? $page : 1;
+            $offset = ( $page - 1 ) * $limit;
+
+            $limits       = 'LIMIT %d, %d';
+            $query_args[] = $offset;
+            $query_args[] = $limit;
+        }
+
+        $cache_group = 'reverse_withdrawal';
+        $cache_key   = 'get_transactions_' . md5( wp_json_encode( $args ) );
+        if ( is_numeric( $args['vendor_id'] ) ) {
+            $cache_group = "reverse_withdrawal_{$args['vendor_id']}";
+        } elseif ( ! $this->is_empty( $args['vendor_id'] ) && 1 === count( $args['vendor_id'] ) ) {
+            $cache_group = "reverse_withdrawal_{$args['vendor_id'][0]}";
+        }
+
+        $data = Cache::get( $cache_key, $cache_group );
+
+        if ( in_array( $args['return'], [ 'vendor_transaction' ], true ) && false === $data ) {
```

Comment:
Is `in_array` needed here? `===` would do.

Reply:
in_array() is needed here, there would be multiple return types and the  actual code will be based on that return  type, `false === $data` is checking the cache value only